packages/backend-defaults/src/entrypoints/rootHttpRouter/rootHttpRouterServiceFactory.ts

```diff
@@ -140,21 +125,34 @@ const rootHttpRouterServiceFactoryWithOptions = (
           app.use(middleware.compression());
           app.use(middleware.logging());
           app.use(healthRouter);
-          app.use(lifecycleMiddleware);
           app.use(routes);
           app.use(middleware.notFound());
           app.use(middleware.error());
         },
       });
 
+      let serverShutdownDelay: HumanDuration = { seconds: 0 };
+
+      if (config.has('backend.lifecycle.serverShutdownDelay')) {
+        serverShutdownDelay = readDurationFromConfig(config, {
+          key: 'backend.lifecycle.serverShutdownDelay',
+        });
+      }
+
+      lifecycle.addBeforeShutdownHook(async () => {
+        const timeoutMs = durationToMilliseconds(serverShutdownDelay);
+        return await new Promise(resolve => {
+          setTimeout(resolve, timeoutMs);
+        });
+      });
```

Comment:
While at it it feels like it makes sense to move this within the `if` block above and make `serverShutdownDelay` a `const`